front/components/agent_builder/capabilities/mcp/utils/dialogUtils.ts

```diff
@@ -0,0 +1,284 @@
+import type { UseFormReturn } from "react-hook-form";
+
+import type { MCPFormData } from "@app/components/agent_builder/AgentBuilderFormContext";
+import type {
+  AgentBuilderAction,
+  ConfigurationPagePageId,
+} from "@app/components/agent_builder/types";
+import { CONFIGURATION_DIALOG_PAGE_IDS } from "@app/components/agent_builder/types";
+import { getMcpServerViewDisplayName } from "@app/lib/actions/mcp_helper";
+import type { MCPServerViewType } from "@app/lib/api/mcp";
+
+import type { DialogMode, SelectedTool } from "../MCPServerViewsDialog";
+
+export interface ModeState {
+  isEditMode: boolean;
+  isInfoMode: boolean;
+  isConfigureMode: boolean;
+  isAddMode: boolean;
+}
+
+export function getModeState(mode: DialogMode | null): ModeState {
+  const isEditMode = !!mode && mode.type === "edit";
+  const isInfoMode = !!mode && mode.type === "info";
+  const isConfigureMode = !!mode && mode.type === "configure";
+  const isAddMode = !mode || mode.type === "add";
+
+  return { isEditMode, isInfoMode, isConfigureMode, isAddMode };
+}
```

Comment:
Can we have a composition of those? Meaning is there a scenario where `isEditMode = true` and `isAddMode = true` make sense?
If no, couldn't we use your `DialogMode.type` directly? Seems to do the job nicely

Reply:
Well I can do something like `const currentMode = mode?.type ?? "add";`

Reply:
But found a solution, this should do :)